base/math.jl

```diff
@@ -128,15 +128,17 @@ end
 @inline function exp2(x::Base.BitInteger)
     if x > 1023
         Inf64
-    elseif x < -1074
-        Float64(0.0)
     elseif x <= -1023
-        # Result will be a subnormal number
-        reinterpret(Float64, Int64(1) << (x + 1074))
+        if x < -1074
+            # Result will be a subnormal number
+            reinterpret(Float64, (1 % Int64) << (x + 1074))
+        else #-1073 < x <= -1023
+            0.0
+        end
     else
         # If x is a Int128, and is outside the range of Int64, then it is not -1023<x<=1023
         # We will cast everything to Int64 to avoid errors in case of Int128
```

Comment:
Since you cast everything to `Int64`, you could also restrict the argument type to `Int`, and have a generic version that accepts `Integer` and calls the main function with argument converted to `Int`.